plugins/destination/plugin_testing.go

```diff
@@ -101,10 +101,24 @@ func getTestLogger(t *testing.T) zerolog.Logger {
 type NewPluginFunc func() *Plugin
 
 type PluginTestSuiteRunnerOptions struct {
-	IgnoreNullsInLists bool // strip nulls from lists before checking equality. Destination setups that don't support nulls in lists should set this to true.
+	// IgnoreNullsInLists allows stripping null values from lists before comparison.
+	// Destination setups that don't support nulls in lists should set this to true.
+	IgnoreNullsInLists bool
+
+	// AllowNull is a custom func to determine whether a data type may be correctly represented as null.
+	// Destinations that have problems representing some data types should provide a custom implementation here.
+	// If this param is empty, the default is to allow all data types to be nullable.
+	AllowNull func(arrow.DataType) bool
+
 	schema.TestSourceOptions
 }
 
+func WithTestSourceAllowNull(allowNull func(arrow.DataType) bool) func(o *PluginTestSuiteRunnerOptions) {
```

Comment:
I would do

```go
type AllowNullFunc func(arrow.DataType) bool
```

and then this becomes

```go
func WithTestSourceAllowNullFunc(allowNullFunc AllowNullFunc) func(o *PluginTestSuiteRunnerOptions)
```

Then we can just add

```go
func WithTestSourceAllowNulls(...arrow.DataType) func(o *PluginTestSuiteRunnerOptions)
```

Which can use a generic `slice.Contains[arrow.DataType]` inside. Could also add a generator function that takes an `...arrow.DataType` and returns a bool-returning contains function but it would be unnecessary if we're already using a generic contains fn.

Reply:
The `slice.Contains` requires comparable types for the slice, and the interfaces aren't.

I don't mind extracting `AllowNullFunc` as type, it that's a strong opinion.